From mibru@gmx.de  Tue Mar 16 13:27:47 2010
From: Michael Brunner <mibru@gmx.de>
Date: Wed, 10 Mar 2010 23:26:37 +0100
Subject: USB: cp210x: Remove double usb_control_msg from cp210x_set_config
To: craig@microtron.org.uk, gregkh@suse.de
Message-ID: <20100310232637.7e14e773@mail.gmx.de>


This patch removes a double usb_control_msg that sets the cp210x
configuration registers a second time when calling cp210x_set_config.
For data sizes >2 the second write gets corrupted.
The patch has been created against 2.6.34-rc1, but all cp210x driver
revisions are affected.

Signed-off-by: Michael Brunner <mibru@gmx.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/cp210x.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/usb/serial/cp210x.c
+++ b/drivers/usb/serial/cp210x.c
@@ -313,11 +313,6 @@ static int cp210x_set_config(struct usb_
 		return -EPROTO;
 	}
 
-	/* Single data value */
-	result = usb_control_msg(serial->dev,
-			usb_sndctrlpipe(serial->dev, 0),
-			request, REQTYPE_HOST_TO_DEVICE, data[0],
-			0, NULL, 0, 300);
 	return 0;
 }
 
